File: 003/solution.c
int lengthOfLongestSubstring(char* s) {
    int i = 0;
    int j;
    int result = 0;
    bool seen[256];
    while (s[i] != '\0') {
        memset(seen, false, sizeof(seen));
        j = i;
        while (s[j] != '\0') {
            if (seen[s[j]]) {
                break;
            } else {
                seen[s[j]] = true;
                j++;
                result = result < j-i? j-i : result;
            }
        }
        i++;
    }
    return result;
}

File: 003/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        result = 0
        for i in range(len(s)):
            j = i
            while len(set(s[i:j+1])) == j+1-i and j < len(s):
                j += 1
            if result < j-i:
                result = j-i
            if len(s) - (i+1) <= result:
                break
        return result
